paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -157,17 +157,23 @@ public class PaparazziPlugin : Plugin<Project> {
         task.compileSdkVersion.set(android.compileSdkVersion())
         task.projectResourceDirs.set(
           project.provider {
-            val resourcesComputer = variant.mergeResourcesProvider?.get()?.resourcesComputer
-            val generateResValuesDirs = resourcesComputer?.generatedResOutputDir ?: project.files()
-            val extraGeneratedResDirs = resourcesComputer?.extraGeneratedResFolders ?: project.files()
-
-            extraGeneratedResDirs.map(projectDirectory::relativize) + localResourceDirs.relativize(projectDirectory) + generateResValuesDirs.map(projectDirectory::relativize)
-          }
+            val resourcesComputer = variant.mergeResourcesProvider
+            val extraGeneratedResDirs =
+              resourcesComputer?.map { it.resourcesComputer.extraGeneratedResFolders }
+                ?: project.provider { project.files() }
+            val generatedResOutputDirs =
+              resourcesComputer?.map { it.resourcesComputer.generatedResOutputDir }
+                ?: project.provider { project.files() }
+            extraGeneratedResDirs
+              .zip(project.provider { localResourceDirs }) { t, u -> t.plus(u) }
+              .zip(generatedResOutputDirs) { t, u -> t.plus(u) }
+              .flatMap { it.relativize(projectDirectory) }
+          }.get()
```

Comment:
This looks fishy, why create a `.provider { }` just to `.get()` it? Am I reading this wrong?

Reply:
Solely to keep this block of code together instead of inlining the respective parts above; it's going away soon anyway, so I took a shortcut

Reply:
stdlib `run { }`?